api/openapi.yaml

```diff
@@ -2047,13 +2047,54 @@ components:
             $ref: '#/components/schemas/AssetScan'
           readOnly: true
 
-    ResourceCleanupState:
-      type: string
-      enum:
-        - Pending
-        - Skipped
-        - Failed
-        - Done
+    ResourceCleanupStatus:
+      type: object
+      required:
+        - state
+        - reason
+        - lastTransitionTime
+      properties:
+        state:
+          description: |
+            Describes the state of resource cleanup.
+            
+            | State   | Description                                                |
+            | ------- | ---------------------------------------------------------- |
+            | Pending | Initial state for cleaning up resources                    |
+            | Skipped | Resource cleanup has been skipped due to Delete Job Policy |
+            | Failed  | Cleaning up resources has been failed                      |
+            | Done    | Resources have been successfully cleaned up                |
+          type: string
+          enum:
+            - Pending
+            - Skipped
+            - Failed
+            - Done
+        reason:
+          description: |
+            Machine readable reason for state transition.
+
+            | State   | Reason          | Description                                                |
+            | ------- | --------------- | ---------------------------------------------------------- |
+            | Pending | AssetScanCreate | AssetScan created                                          |
+            | Skipped | DeletePolicy    | Resource cleanup has been skipped due to Delete Job Policy |
+            | Failed  | ProviderError   | Failed due to Provider error                               |
+            | Failed  | InternalError   | Failed due to internal error                               |
```

Comment:
do we have a cleanup timeout? I can't remember :/

Reply:
There is no timeout for resource cleanup. We might add a timeout to the reconcile logic at some point, but I do not think it is critical for now.

Reply:
👍 cool, if we had it then TimedOut would be another reason to have here for failed. If we don't have a timeout then we're good.